pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -612,6 +643,22 @@ func (a *FlavorAssigner) findFlavorForPodSetResource(
 		// Calculate representativeMode for this assignment as the worst mode among all requests.
 		representativeMode := fit
 		for rName, val := range requests {
+			// Ensure the same resource flavor is used for the workload slice as in the original admitted slice.
+			if a.preemptWorkloadSlice != nil {
```

Comment:
please also guard with the new feature gate.

Reply:
done

Reply:
please mark as resolved